File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
words = 1
sentences = 0
spaces = 0

for i in range(len(text)):
    if (text[i] >= "a" and text [i] <= "z") or (text[i] >= "A" and text [i] <= "Z"):
        letters += 1

for i in range(len(text)):
    if text[i] == " ":
        spaces += 1

for q in range(len(text)):
    if text[q] == " " and text[q + 1] != " ":
        words += 1

for a in range(len(text)):
    if text[a] == "." or text[a] == "!" or text[a] == "?":
        sentences += 1

L = float((letters * 100) / words)
S = float((sentences * 100) / words)

grade = round(float(0.0588 * L - 0.296 * S - 15.8))

if grade <= 1:
    print("Before Grade 1\n")

if grade > 1 and grade < 16:
    print(f"Grade {grade} \n")

if grade >= 16:
    print("Grade 16+\n")

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{

    string text = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int i = 0, n = strlen(text); i <= n; i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters = letters + 1;
        }
    }

    for (int q = 0, w = strlen(text); q <= w; q++)
    {
        if (text[q] == ' ' && text[q + 1] != ' ')
        {
            words = words + 1;
        }
    }

    for (int a = 0, z = strlen(text); a <= z; a++)
    {
        if (text[a] == '.' || text[a] == '!' || text[a] == '?')
        {
            sentences = sentences + 1;
        }
    }

    float L = (letters * 100) / words;
    float S = (sentences * 100) / words;

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);


    // printf("%i letter(s)\n", letters);
    // printf("%i word(s)\n", words);
    // printf("%i sentence(s)\n", sentences);
    // printf("%i\n", grade);

    if (grade <= 1)
    {
        printf("Before Grade 1\n");
    }

    if (grade > 1 && grade < 16)
    {
        printf("Grade %i\n", grade);
    }

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }

}
